code/game/objects/items/weapons/tanks/watertank.dm

```diff
@@ -248,53 +258,55 @@
 		loc = tank
 	return
 
-/obj/item/extinguisher/mini/nozzle/Move()
+/obj/item/extinguisher/nozzle/Move()
 	..()
 	if(tank && loc != tank.loc)
 		loc = tank
```

Comment:
this entire proc worries me, NODROP may just be a better option here.

Reply:
I don't think this proc is actually required as the new version should automatically set its location to the tank, addressing if there is no tank can be handled in the dropped proc too